lib/facility.rb

```diff
@@ -1,14 +1,82 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services,
+              :registered_vehicles
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  attr_accessor :collected_fees
+
+
+  def initialize(facility_details)
+    @name = facility_details[:name]
+    @address = facility_details[:address]
+    @phone = facility_details[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if @services.include?('Vehicle Registration')
+      vehicle.registration_date = Date.today.year
+      if vehicle.antique?
+        @collected_fees += 25
+        vehicle.plate_type = :antique
+      elsif vehicle.electric_vehicle?
+        @collected_fees += 200
+        vehicle.plate_type = :ev
+      else 
+        @collected_fees += 100
+        vehicle.plate_type = :regular
+      end
+      @registered_vehicles << vehicle
+    else
+      nil
+    end
+  end
+
+  def administer_written_test(registrant)
+    if @services.include?('Written Test')
+      true
```

Comment:
Not sure why this 'true' is here. 
To reduce nesting, you can string together your if conditions, for example, your line 44 could read: `if @services.include?("Written Test") && registrant.permit? && registrant.age >= 16`

Reply:
Same feedback for the two following methods